bindings/azure/blobstorage/metadata.yaml

```diff
@@ -0,0 +1,112 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: azure.blobstorage
+version: v1
+status: stable
+title: "Azure Blob Storage"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/blobstorage/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create blob."
+    - name: get
+      description: "Get blob."
+    - name: delete
+      description: "Delete blob."
+    - name: list
+      description: "List blob."
+capabilities: []
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+    - name: accountName
+      required: true
+      sensitive: false
+      description: "The storage account name"
+      example: '"mystorageaccount"'
+authenticationProfiles:
+  - title: "Connection string"
+    description: "Authenticate using a connection string."
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: "Shared access policy connection string for Blob Storage."
+        example: '"BlobEndpoint=https://storagesample.blob.core.windows.net;SharedAccessSignature={KeySig}"'
+        # If omitted, uses the same values as "<root>.binding"
+        binding:
+          output: true
+  - title: "Account Key"
+    description: |
+      Authenticate using a pre-shared "account key".
+    metadata:
+      - name: accountKey
+        required: true
+        sensitive: true
+        description: "The key to authenticate to the Storage Account."
+        example: '"my-secret-key"'
+      - name: accountName
+        required: true
+        sensitive: false
+        description: "The storage account name"
+        example: '"mystorageaccount"'
+metadata:
+  - name: containerName
+    description: "The name of the container to be used for Dapr state. The container will be created for you if it doesn't exist."
+    required: true
+    example: '"container"'
+  - name: azureEnvironment
+    description: Name of the Azure Environment if using a different Azure cloud.
+    example: '"AZUREPUBLICCLOUD"'
+  - name: endpoint
+    description: | 
+      Optional custom endpoint URL. This is useful when using the Azurite emulator or when using custom domains for Azure Storage (although this is not officially supported). 
+      The endpoint must be the full base URL, including the protocol (http:// or https://), the IP or FQDN, and optional port.
+    example: '"http://127.0.0.1:10000"'
+    type: string
+  - name: contentType
+    description: "The blob's content type."
+    example: '"text/plain"'
+  - name: contentMD5
+    description: "The blob's MD5 hash."
+    example: '"vZGKbMRDAnMs4BIwlXaRvQ=="'
```

Comment:
I haven't checked bindings code but given there is **a lot** of shared code: have you checked if these fields make sense in the component metadata? Because they are documented as being component metadata settings but they are not even mentioned in the state store and when I searched in the bindings code it appeared they are request-based metadata, not a component settings one.